autogluon/scheduler/hyperband.py

```diff
@@ -236,6 +241,16 @@ def __init__(self, train_fn, args=None, resource=None,
         # This lock protects both _running_tasks and terminator, the latter
         # does not define its own lock
         self._hyperband_lock = mp.Lock()
+        if resume:
```

Comment:
shall we move this code block before parent init with resume being the validated boolean?

Reply:
The resume block, calling self.load_state_dict, needs to be down here, after everything else has been created. This is because of the way that load_state_dict is done now.

Previously, it relied in pickling complex objects like the searcher. Now, it works differently. The checkpoint only stores what I call "mutable state" of the searcher, and of other objects which would not pickle well. And load_state_dict is first running __init__ for all these objects, and then just puts in their resp mutable state from the checkpoint.

This is the reason I had to move the resume block here, and also that we cannot just pass resume to FIFOScheduler, because it could not do the job properly.

Reply:
This sounds a little complex, but we need that for our more complex searchers we will contribute. Note that the default for any new searcher is just what has been done before, namely pickling the whole searcher object.